src/main/java/me/ryanhamshire/GPFlags/flags/FlagDef_NoIceMelt.java

```diff
@@ -0,0 +1,50 @@
+package me.ryanhamshire.GPFlags.flags;
+
+import me.ryanhamshire.GPFlags.*;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.block.BlockFadeEvent;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class FlagDef_NoIceMelt extends FlagDefinition {
+
+    public FlagDef_NoIceMelt(FlagManager manager, GPFlags plugin) {
+        super(manager, plugin);
+    }
+
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void onGrowth(BlockFadeEvent event) {
+        Block block = event.getBlock();
+
+        Flag flag = this.getFlagInstanceAtLocation(block.getLocation(), null);
+        if (flag == null) return;
+
+        if (event.getBlock().getType() != Material.ICE) return;
+        event.setCancelled(true);
+    }
+
+    @Override
+    public String getName() {
+        return "NoIceMelt";
+    }
+
+    @Override
+    public MessageSpecifier getSetMessage(String parameters) {
+        return new MessageSpecifier(Messages.EnableNoIceMelt);
+    }
+
+    @Override
+    public MessageSpecifier getUnSetMessage() {
+        return new MessageSpecifier(Messages.DisableNoIceMelt);
+    }
+
+    @Override
+    public List<FlagType> getFlagType() {
+        return Arrays.asList(FlagType.CLAIM, FlagType.WORLD, FlagType.SERVER);
```

Comment:
For both of these, Im really REALLY iffy about this being a world/server flag.
These events are going to be called an awful lot, and are just going to lag the server.

I recommend using just CLAIM for the flag type.